packages/dom-adapters/src/BlockToolAdapter/index.ts

```diff
@@ -171,39 +207,70 @@ export class BlockToolAdapter {
       return;
     }
 
-    const action = event.detail.action;
+    if (isNativeInput) {
+      const currentElement = input as HTMLInputElement | HTMLTextAreaElement;
+      const start = currentElement.selectionStart!;
+      const end = currentElement.selectionEnd!;
+
+      const action = event.detail.action;
+
+      switch (action) {
```

Comment:
I'm a little worried about this part of logic that doesn't use the `caretAdapter` class. Because onUpdate callback cannot get any information about the update from caretEvent

Reply:
Agree, would be better if CaretAdapter handles native inputs selection on its own. I guess you can leave a to-do here to consider that for caret adapter re-implementation